src/libvirt-xml-update.js

```diff
@@ -56,7 +56,7 @@ export function updateBootOrder(domXml, devices) {
     const doc = getDoc(domXml);
     const domainElem = doc.firstElementChild;
     if (!domainElem)
-        throw new Error("updateBootOrder: domXML has no domain element");
+        Promise.reject(new Error("updateBootOrder: domXML has no domain element"));
```

Comment:
This and the function above isn't a promise so this looks wrong to me? So I think `domainChangeBootOrder` should instead handle a possible exception from `updateBootOrder`

Reply:
It is handled one layer above, at  [BootOrderModal](https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/vm/overview/bootOrder.jsx#L255), which calls `domainChangeBootOrder`. This way, this exception bubbles all the way to the UI and is shown as an error in the dialog.

Reply:
Yes, but it feels kinda weird to me, if you want to use `updateBootOrder` somewhere where there is no promise it simply won't work.

Reply:
From what I have read and tried locally, `throw` and `Promise.reject` have the same behaviour when outside of promise chain. Meaning in both cases, they will cause Uncaught exception.
The only difference is that if you re inside a promise chain, `Promise.reject` can be caught by `Promise.catch`, while `throw` can't.